clients/src/main/java/org/apache/kafka/clients/telemetry/DefaultClientTelemetry.java

```diff
@@ -436,26 +444,26 @@ else if (state == TelemetryState.terminating_push_in_progress)
      *  - UnknownSubscription or Unsupported Compression: Retry
      *  After a new subscription is created, the current state is transitioned to push_needed to wait for another push.
      *
-     * @param data pushTelemetryResponseData
-     * @throws IllegalTelemetryStateException when subscription is mull
+     * @param errorCode response body error code.
+     * @throws IllegalTelemetryStateException when subscription is null.
      */
-    private void handlePushTelemetryResponseDataErrors(final PushTelemetryResponseData data) {
+    private void handleResponseErrorCode(short errorCode) {
         TelemetrySubscription currentSubscription = subscription().orElseThrow(
         () -> new IllegalTelemetryStateException(String.format("Subscription cannot be null.  Aborting.")));
 
         final long retryMs;
         // We might want to wait and retry or retry after some failures are received
-        if (isAuthorizationFailedError(data.errorCode())) {
+        if (isAuthorizationFailedError(errorCode)) {
             retryMs = 30 * 60 * 1000;
-            log.warn("Error code: {}. Reason: Client is permitted to send metrics.  Retry automatically in {}ms.", data.errorCode(), retryMs);
+            log.warn("Error code: {}. Reason: Client is permitted to send metrics.  Retry automatically in {}ms.", errorCode, retryMs);
```

Comment:
I'm not sure what "Client is permitted to send metrics" means. Should it be client is _not_ allowed?

Reply:
Any chance I could get a space between before `ms` in the message?

Reply:
🤦 intended to write Client is not permitted...